### DISCUSSION TITLE
$parent.function() is working on wire.click but not from Alpine variables

### DISCUSSION THREAD

**Discussion by joffreypersia** [livewire/livewire#7622] [4. General]
Hi, 

I am building a multi-step form. Each step has a form and is built like : 
```blade
<div
    x-data="{
        init() {
            this.$watch('parameters', value => {
                console.log('parameters modified watched from step 1');
            });
        },
        modifyParam() {
            this.parameters[{{$currentStep}}] = 'Hello';
            console.log(this.parameters);
        },
        submit(){
            this.isLoading = true;
            this.modifyParam();
            // go to next step
            $parent.nextStep();
        },
    }"
>
    <form
        id="stepForm{{$currentStep}}"
        @submit.prevent="submit()"
    >

        <p>On the step : {{ $currentStep }}</p>

        {{ var_dump($parameters) }}

        <button wire:click="$parent.nextStep()">Next from child</button>
    </form>
</div>
```

The button submit is on the parent layout and triggers the form with the form ID

I noticed that $parent works on the wire:click but not on the function submit(). The problem is that I am using submit() to make all the checks on each step. 

Is there a way to use $parent inside an alpine function ? because I would like to not use $displatch as it creates an "update" unnecessary. 

Thank you,

**Comment by joffreypersia** [comment]
I found a "workaround" like so : 
```js
submit(){
      this.isLoading = true;
      this.modifyParam();
      // go to next step
      this.$dispatch('next-step-event');
}
```

And the parent component has : 
```js
init() {
      $wire.on('next-step-event', () => {
          $wire.nextStep();
      });
}
```

I read on the doc that dispatching events client-side is, by default, a better option for performance.

What do you think ?

**Comment by N1ebieski** [comment]
Replace wire:click with x-on:click and $parent with $wire.$parent and everything works > https://wirebox.app/b/4mpkv

**Reply by joffreypersia** [reply]
Oh that's perfect !!! Thank you ! 🫶

**Comment by joffreypersia** [comment]
> Replace wire:click with x-on:click and $parent with $wire.$parent and everything works > https://wirebox.app/b/4mpkv
